### PR TITLE
Parallelize EQLs predictions and Jacobian build

### PR DESCRIPTION
**Pull request by santisoler**
Parallelize the build of the Jacobian matrix and the prediction step on
equivalent layer gridders through `numba.prange` iterators. This enables the
`fit` and `predict` methods to use all the available CPUs, greatly reducing the
computational time. The `prange` is only applied to the outer loops for each
task, i.e. distributing computation points between CPUs, but leaving the
iteration through the sources on single ones.

**Reminders**:

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst` and the base `__init__.py` file for the package.
- [ ] Write detailed docstrings for all functions/classes/methods. It often helps to design better code if you write the docstrings first.
- [ ] If adding new functionality, add an example to the docstring, gallery, and/or tutorials.
- [ ] Add your full name, affiliation, and [ORCID](https://orcid.org) (optional) to the `AUTHORS.md` file (if you haven't already) in case you'd like to be listed as an author on the [Zenodo](https://zenodo.org/communities/fatiando) archive of the next release.

### CODE REVIEW

**Review by leouieda — APPROVED** [review]
:+1: but we need to be careful when combing this with Dask to turn off the numbs parallelism. It would be interesting to have a Dask delayed version of  `grid` so we can turn this off for cross-validation but then still have parallel gridding.

**Comment by santisoler** [comment]
Yes! I was thinking the same thing and wanted to test how Dask + Numba parallelization works. Maybe there's no need for switching on and off the Numba `parallel=True` and Dask assigns a single CPU to each Numba job :thinking:

**Comment by santisoler** [comment]
I've just run this gridding notebook https://github.com/fatiando/transform2020-lightning/tree/master/santi with the `EQLHarmonic` from Harmonica v0.1.0 and with the parallel version from this PR. The cross validation (using dask) take the same time on both cases, while the prediction on a regular grid is obviously faster when parallel (more than half of the time). So I suspect that Dask might be smart enough to know how to handle Numba parallelizations. If anyone has better insight on this one, feel free to jump in!

**Comment by leouieda** [comment]
Dask isn't smart enough but I think the problems are small enough not to show up. But what happens is that the CPU cores are over subscribed so we may not get the best performance possible. But we can turn off numba parallelism with an environment variable. So I think this is safe to merge. We can add more support for dask when we see problems arise.

File: harmonica/equivalent_layer/utils.py [hunk 1]
```diff
@@ -28,19 +28,21 @@ def jacobian_numba(coordinates, points, jac, greens_function):
...
+    It works both for Cartesian and spherical coordiantes. We need to pass the
+    corresponding Green's function through the ``greens_function`` argument.
+    The prediction is run in parallel in order to reduce the computation time.
     """
     east, north, upward = coordinates[:]
     point_east, point_north, point_upward = points[:]
-    for i in range(east.size):
-        for j in range(point_east.size):
+    for i in prange(east.size):
+        for j in prange(point_east.size):
```
Comment (leouieda) [inline harmonica/equivalent_layer/utils.py:45]: Might be worth benchmarking if it's better to have a `prange` in the outer `for` only.

Comment (santisoler) [reply harmonica/equivalent_layer/utils.py]: I haven't seen any significant difference, but I think it's wiser to leave only `prange` in the outer loop only, we will never have that many cores to split the computation of almost one matrix element per core (remind me of reading this same sentence in 30 years from now haha).

**Comment by santisoler** [comment]
> Dask isn't smart enough but I think the problems are small enough not to show up. But what happens is that the CPU cores are over subscribed so we may not get the best performance possible. But we can turn off numba parallelism with an environment variable. So I think this is safe to merge. We can add more support for dask when we see problems arise.

Ok. I don't have too much experience with parallelization and Dask is surprisingly difficult for me, probably I need to spend more time with it. So my thoughts are:
1. Consider first the regular user experience: most user will have several CPUs on their computers (even the low-profile computers nowdays have more than one core), so I would say that every CPU demanding task in Harmonica should leverage the available cores to reduce computation time without hassle. I think `prange` fills that gap.
2. We should also leave the possibility of more complex parallelization of the code through Dask. So if we can turn off Numba parallelization with an environment variable, I think the jitted functions won't need any modification to turn parallelization on and off.

**Comment by leouieda** [comment]
👍🏽 I agree with both points 🙂

**Pull request merged**